tool_collections/cufflinks/gffread/gffread.xml

```diff
@@ -0,0 +1,425 @@
+                <option value="-B">(-B with -V) single-exon transcripts are also checked on the opposite strand</option>
+            </param>
+        </xml>
+        <xml name="trackname">
+            <param name="tname" type="text" value="" size="30" optional="true" label="(-t) Trackname to use in the second column of each GFF output line">
+                <validator type="regex">\w+</validator>
+            </param>
+        </xml>
+        <xml name="merge_opts">
+             <option value="-K">(-K) also collapse shorter, fully contained transcripts with fewer introns than the container</option>
+             <option value="-Q">(-Q) remove the containment restriction (multi-exon transcripts will be collapsed if just their introns match, while single-exon transcripts can partially overlap 80%)</option>
+             <option value="-d dupinfo">(-d) output collapsing info</option>
+        </xml>
+        <xml name="cluster_opts">
+             <option value="--force-exons">(--force-exons) make sure that the lowest level GFF features are printed as 'exon' features</option>
+             <option value="-Z">(-Z) merge close exons into a single exon (for intron size &lt; 4)</option>
+        </xml>
+        <xml name="merge_opt_sel">
+            <param name="merge_options" type="select" display="checkboxes" multiple="true" label="Merge options">
+                <expand macro="cluster_opts" />
+                <expand macro="merge_opts" />
+            </param>
+        </xml>
+        <xml name="cluster_opt_sel">
+            <param name="merge_options" type="select" display="checkboxes" multiple="true" label="Cluster options">
+                <expand macro="cluster_opts" />
+            </param>
+        </xml>
+    </macros>
+    <command>
+<![CDATA[
+    #if $reference_genome.source == 'history':
+        ln -s $reference_genome.genome_fasta genomeref.fa &&
+    #end if
+    gffread $input 
+    #if $reference_genome.source == 'cached':
+        -g "${reference_genome.fasta_indexes.fields.path}"
+        #if $reference_genome.ref_filtering and str($reference_genome.ref_filtering) != '':
+            #echo ' '.join(str($reference_genome.ref_filtering).split(','))
+        #end if
+    #elif $reference_genome.source == 'history':
+        -g genomeref.fa
+        #if $reference_genome.ref_filtering and str($reference_genome.ref_filtering) != '':
+            #echo ' '.join(str($reference_genome.ref_filtering).split(','))
+        #end if
+    #end if
+    #if $filtering and str($filtering) != '':
+        #echo " "
+        #echo ' '.join(str($filtering).split(','))
+    #end if
+    #if $maxintron and $maxintron > 0:
+        -i $maxintron
+    #end if
+    #if $region.region_filter == 'filter':
+        -r $region.range $region.discard_partial
+    #end if
+    #if $merging.merge_sel != 'none':
+        $merging.merge_cmd
+        #echo ' '.join(str($merging.merge_options).split(','))
```

Comment:
I hope to make such statements a little bit simpler with: https://bitbucket.org/galaxy/galaxy-central/pull-request/624/add-__getitem__-to-select-parameters-to